### PR TITLE
Issue 25 help thread system

### LINKED ISSUE #25: Help Thread system
### General Troubleshooting

- [X] I have checked for similar issues on the Issue-tracker.
- [X] I have checked the branches or the maintainers' PRs for upcoming features.

### Feature Request

One channel and you can use a command like /ask to create a thread. only 2 threads can be created per day and threads are archived after 3 days. There is also has to be a channel which lists out all the active threads

### PR DESCRIPTION
**Pull request by RealYusufIsmail**
[code_of_conduct]: https://github.com/Programming-Org/Programming/blob/master/.github/CODE_OF_CONDUCT.md
[new_issue]: https://github.com/Programming-Org/Programming/issues/new/choose

## Pull-request

- [x] I have read the [code of conduct][code_of_conduct].
- [x] I have created a relating [issue][new_issue].

### Changes

- [x] Existing code
- [x] New feature

Closes : #25 

## Description

Added the ability to create a thread channel which will be used for the new help system

## TODO:
- [x] ask command
- [x] Finish active question system
- [x] edit category command
- [x] close command
- [x] Check sync system works

### CODE REVIEW

File: src/main/java/io/github/org/programming/bot/commands/moderation/UnBanCommand.java [hunk 1]
```diff
@@ -65,6 +69,11 @@ private MessageEmbed unbanEmbed(Member member, @NotNull Member moderator,
 
     @Override
     public SlashCommand build() {
-        return null;
+        return new SlashCommandBuilder("unban", "The user to unban")
+            .addOption(OptionType.USER, "member", "The user to unban", true)
+            .addOption(OptionType.STRING, "reason", "The reason for the unban", true)
+            .build()
+            .setBotPerms(Permission.BAN_MEMBERS)
+            .setUserPerms(Permission.BAN_MEMBERS);
```
Comment (Haleloch) [inline src/main/java/io/github/org/programming/bot/commands/moderation/UnBanCommand.java:77]: Why `setUserPerms` ?

File: src/main/java/io/github/org/programming/bot/commands/thread/AskCommand.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    private List<Command.Choice> choices =
+            List.of(new Command.Choice("Java", "java"), new Command.Choice("C++", "c++"),
+                    new Command.Choice("C#", "c#"), new Command.Choice("Python", "python"),
+                    new Command.Choice("JavaScript", "js"), new Command.Choice("PHP", "php"),
+                    new Command.Choice("C", "c"), new Command.Choice("Go", "go"),
+                    new Command.Choice("Rust", "rust"), new Command.Choice("Swift", "swift"),
+                    new Command.Choice("Kotlin", "kotlin"), new Command.Choice("Scala", "scala"),
+                    new Command.Choice("TypeScript", "ts"), new Command.Choice("Other", "other"));
+}
```
Comment (Haleloch) [inline src/main/java/io/github/org/programming/bot/commands/thread/AskCommand.java]: What is this ?

File: src/main/java/io/github/org/programming/bot/commands/moderation/UnBanCommand.java (hunk 1 above)
Comment (RealYusufIsmail) [reply src/main/java/io/github/org/programming/bot/commands/moderation/UnBanCommand.java:77]: UserPerms mean that user using the command needs have ban perms.

Comment (RealYusufIsmail) [reply src/main/java/io/github/org/programming/bot/commands/moderation/UnBanCommand.java:77]: This is to make sure if someone does not have ban perms they can not use this command

File: src/main/java/io/github/org/programming/bot/commands/thread/AskCommand.java (hunk 2 above)
Comment (RealYusufIsmail) [reply src/main/java/io/github/org/programming/bot/commands/thread/AskCommand.java]: when asking you do ```/ask name topic``` The topic will be used to differentiate in the active question channel

Comment (RealYusufIsmail) [reply src/main/java/io/github/org/programming/bot/commands/thread/AskCommand.java]: topic will be a list of choices

Comment (RealYusufIsmail) [reply src/main/java/io/github/org/programming/bot/commands/thread/AskCommand.java]: Screenshot 2022-08-13 at 11 28 27
something like this

Comment (RealYusufIsmail) [reply src/main/java/io/github/org/programming/bot/commands/thread/AskCommand.java]: might rename it to category makes more sense

File: src/main/java/io/github/org/programming/bot/commands/thread/ActiveQuestionsHandler.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            .concat("**Kotlin**: ")
+            .concat("\n")
+            .concat("\n")
+            .concat("**Ts**: ")
+            .concat("\n")
+            .concat("\n")
+            .concat("**Other**: ")
+            .concat("\n")
+            .concat("\n");
+    }
```
Comment (Haleloch) [inline src/main/java/io/github/org/programming/bot/commands/thread/ActiveQuestionsHandler.java]: we could perhaps factorize the code
and store the languages/frameworks in a single place (an array for example?)

File: src/main/java/io/github/org/programming/bot/commands/thread/AskCommand.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    public static List<Command.Choice> categoryChoices =
+            List.of(new Command.Choice("Java", "java"), new Command.Choice("C++", "c++"),
+                    new Command.Choice("C#", "c#"), new Command.Choice("Python", "python"),
+                    new Command.Choice("JavaScript", "js"), new Command.Choice("PHP", "php"),
+                    new Command.Choice("C", "c"), new Command.Choice("Go", "go"),
+                    new Command.Choice("Rust", "rust"), new Command.Choice("Swift", "swift"),
+                    new Command.Choice("Kotlin", "kotlin"), new Command.Choice("Scala", "scala"),
+                    new Command.Choice("TypeScript", "ts"), new Command.Choice("Other", "other"));
```
Comment (Haleloch) [inline src/main/java/io/github/org/programming/bot/commands/thread/AskCommand.java]: same ?

File: src/main/java/io/github/org/programming/bot/commands/thread/ActiveQuestionsHandler.java (hunk 3 above)
Comment (Haleloch) [reply src/main/java/io/github/org/programming/bot/commands/thread/ActiveQuestionsHandler.java]: Done with commit b63fdb3d7d697582b4d85c4f335107fb82b6a743

**Review by Haleloch — APPROVED** [review]
OK

**Pull request merged**